fujiji-client/src/pages/search/index.jsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Box, Center, SimpleGrid, Text, Spinner,
+} from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+import { AdListingCard, withSession } from '../../components';
+import { getAllListings } from '../../server/api';
+
+function SearchPage() {
+  const router = useRouter();
+  const [listings, setListings] = useState([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  // for this page we are using client-side rendering
+  // so we can show loading indicator when fetching huge list of listings
+  useEffect(() => {
+    async function fetchData() {
+      const res = await getAllListings();
+      if (res.error) {
+        setErrorMessage(res.error);
+      } else {
+        setListings(res);
+      }
+    }
+
+    fetchData();
+  }, []);
```

Comment:
can we also remove empty dependency array (`[]`) in `useEffect()`? This is not needed, I don't think.